controllers/integration/integration_adapter.go

```diff
@@ -189,15 +193,24 @@ func (a *Adapter) getAllApplicationComponents(application *hasv1alpha1.Applicati
 }
 
 // getAllApplicationReleaseLinks returns the ReleaseLinks used by the application being processed. If matching
-// ReleaseLinks are not found, an error will be returned.
+// ReleaseLinks are not found, an error will be returned. A ReleaseLink will only be returned if it has the
+// release.appstudio.openshift.io/auto-release label set to true or if it is missing the label entirely.
 func (a *Adapter) getAllApplicationReleaseLinks(application *hasv1alpha1.Application) (*[]releasev1alpha1.ReleaseLink, error) {
 	releaseLinks := &releasev1alpha1.ReleaseLinkList{}
-	opts := []client.ListOption{
-		client.InNamespace(application.Namespace),
-		client.MatchingFields{"spec.application": application.Name},
+	labelSelector := labels.NewSelector()
+	labelRequirement, err := labels.NewRequirement("release.appstudio.openshift.io/auto-release", selection.NotIn, []string{"false"})
+	if err != nil {
+		return nil, err
+	}
+	labelSelector = labelSelector.Add(*labelRequirement)
+
+	opts := client.ListOptions{
+		Namespace:     application.Namespace,
+		FieldSelector: fields.OneTermEqualSelector("spec.application", application.Name),
+		LabelSelector: labelSelector,
 	}
 
-	err := a.client.List(a.context, releaseLinks, opts...)
+	err = a.client.List(a.context, releaseLinks, &opts)
```

Comment:
Wondering if it makes sense to move the `&` to the line where the instance is created. We do always the same for resources declarations.

Reply:
done